app/src/main/java/org/mozilla/reference/browser/pip/PictureInPictureFeature.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.mozilla.reference.browser.pip
+
+import android.annotation.TargetApi
+import android.app.Activity
+import android.app.PictureInPictureParams
+import android.os.Build
+import androidx.fragment.app.Fragment
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import org.mozilla.reference.browser.BackHandler
+
+/**
+ * A simple implementation of Picture-in-picture mode support.
+ *
+ * @param sessionManager Session Manager for observing the selected session's fullscreen mode changes.
+ * @param activity the activity with the EngineView for calling PIP mode when required; the AndroidX Fragment
+ * doesn't support this.
+ * @param fragment the current fragment for back button handling.
+ */
+class PictureInPictureFeature(
+    sessionManager: SessionManager,
+    private val activity: Activity?,
+    private val fragment: Fragment
+) : SelectionAwareSessionObserver(sessionManager), LifecycleAwareFeature {
```

Comment:
nit: Extended `SelectionAwareSessionObserver` exposes a large public API since that class extends `Session.Observer` and `SessionManager.Observer`. For that reason I usually prefer to keep an internal instance of that class instead of extending it.

Reply:
That's a good point since I don't even need to be selection aware (also removing the need to be lifecycle aware)!